security/src/test/groovy/io/micronaut/security/authorization/AuthorizationSpec.groovy

```diff
@@ -318,6 +376,42 @@ class AuthorizationSpec extends EmbeddedServerSpecification {
         }
     }
 
+    @Requires(property = 'spec.name', value = 'AuthorizationSpec')
+    @Controller('/subtypeargumentbinder')
+    @Secured("isAuthenticated()")
+    static class PrincipalSubtypeArgumentBinderController {
+
+        @Get("/single-server-authentication")
+        @SingleResult
+        Publisher<String> singleServerAuthentication(@User ServerAuthentication authentication) {
```

Comment:
I agree. I wasn't sure if the original spec author was explicitly trying to make sure it worked with `Publisher` for some non-obvious reason since it was explicitly being called out in the spec method descriptions.